### ISSUE TITLE
Mitigate scenarios where pod deployments flap

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#7149]
**Rancher Version:**
1.2.1

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
multi-node

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
kubernetes

**Steps to Reproduce:**
1. If k8s gets into a scenario where it is rapidly deploying and destroying a lot of pods, rancher starts to fall further and further behind in event processing. Eventually, Rancher process can become starved and important events don't get processed.

**Results:** 
The process_instance table gets full (thousands of rows) of SCHEDULED but not completed entries.

**Expected:**
We mitigate so thatwe are not flooding cattle with events. There is no single correct behavior or solution. We just need to find ways to mitigate.

**Comment by aemneina** [comment]
@cjellick not sure if this is related, or configurable... or even helpful. https://github.com/kubernetes/kubernetes/issues/12648 Maybe there's a setting where we can make fail/retry more conservative.

**Comment by sangeethah** [comment]
Tested with v1.2.2-rc3

On a kubernetes environment:

Create the following pods all at 1 time:

1. 100 Replication controllers with scale 2 such that the pods would die within 5 secs of starting using the following yml:
```
apiVersion: v1
kind: ReplicationController
metadata:
  name: invalidrc-4
spec:
  replicas: 2
  selector:
    name: invalidrc-4
  template:
    metadata:
      labels:
        name: invalidrc-4
    spec:
      containers:
        - name: invalidrc-4
          image: sangeetha/testnewhostrouting
          command: ["/bin/sh","-c"]
          args: ["sleep 5; /bin/false"]
          ports:
            - containerPort: 80
```

2. Another RC with valid image (which deploys correctly) with scale 20.

I am seeing containerevent.create  process go up as much as 250 . We expect it grow up to a max of 75 (or a max of 150).


```
mysql> select count(process_name), process_name from process_instance where end_time is null group by process_name;
+---------------------+-----------------------------+
| count(process_name) | process_name                |
+---------------------+-----------------------------+
|                 248 | containerevent.create       |
|                  42 | image.create                |
|                 104 | image.purge                 |
|                  17 | instance.create             |
|                   6 | instance.remove             |
|                  22 | instance.stop               |
|                   1 | instancelink.purge          |
|                   3 | ipaddress.purge             |
|                 277 | mount.create                |
|                  67 | mount.deactivate            |
|                   1 | service.update              |
|                   1 | serviceexposemap.create     |
|                 129 | volume.activate             |
|                 150 | volume.create               |
|                  57 | volume.deactivate           |
|                 185 | volume.purge                |
|                  26 | volumestoragepoolmap.remove |
+---------------------+-----------------------------+
17 rows in set (0.01 sec)
```

All the pods with valid image were stuck in  "


only 3 of them out of 20 is got to “Running” state after 6 minutes. Rest of them are stuck at "ContainerCreating" state.



 ```Sangeethas-MacBook-Pro:kubectl41 sangeethahariharan1$ ./kubectl describe pod validrc02-kh231
Name:        validrc02-kh231
Namespace:    default
Node:        sank8sdec221.localdomain/159.203.64.144
Start Time:    Thu, 22 Dec 2016 15:02:00 -0800
Labels:        name=validrc02
Status:        Pending
IP:        
Controllers:    ReplicationController/validrc02
Containers:
  validrc02:
    Container ID:    
    Image:        sangeetha/testnewhostrouting
    Image ID:        
    Port:        80/TCP
    State:        Waiting
      Reason:        ContainerCreating
    Ready:        False
    Restart Count:    0
    Volume Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-pd6lo (ro)
    Environment Variables:    <none>
Conditions:
  Type        Status
  Initialized     True 
  Ready     False 
  PodScheduled     True 
Volumes:
  default-token-pd6lo:
    Type:    Secret (a volume populated by a Secret)
    SecretName:    default-token-pd6lo
QoS Class:    BestEffort
Tolerations:    <none>
Events:
  FirstSeen    LastSeen    Count    From                    SubobjectPath            Type        Reason        Message
  ---------    --------    -----    ----                    -------------            --------    ------        -------
  8m        8m        1    {default-scheduler }                            Normal        Scheduled    Successfully assigned validrc02-kh231 to sank8sdec221.localdomain
  4m        4m        1    {kubelet sank8sdec221.localdomain}    spec.containers{validrc02}    Normal        Pulling        pulling image "sangeetha/testnewhostrouting"
  4m        4m        1    {kubelet sank8sdec221.localdomain}    spec.containers{validrc02}    Normal        Pulled        Successfully pulled image "sangeetha/testnewhostrouting"
  4m        4m        1    {kubelet sank8sdec221.localdomain}    spec.containers{validrc02}    Normal        Created        Created container with docker id 495cd6814a3c; Security:[seccomp=unconfined]
  4m        4m        1    {kubelet sank8sdec221.localdomain}    spec.containers{validrc02}    Normal        Started        Started container with docker id 495cd6814a3c
Sangeethas-MacBook-Pro:kubectl41 sangeethahariharan1$ 
```

```
logs have lot of these exceptions
```2016-12-22 23:27:53,231 INFO  [:] [] [] [] [ecutorService-1] [i.c.p.c.v.impl.ConfigUpdatePublisher]         Finished work item agent:51 [705568664] with exception [class io.cattle.platform.async.utils.TimeoutException:Timeout] 
2016-12-22 23:27:53,240 INFO  [:] [] [] [] [ecutorService-1] [i.c.p.c.v.impl.ConfigUpdatePublisher]         Finished work item agent:51 [188911553] with exception [class io.cattle.platform.async.utils.TimeoutException:Timeout] 
2016-12-22 23:27:53,240 INFO  [:] [] [] [] [ecutorService-1] [i.c.p.c.v.impl.ConfigUpdatePublisher]         Finished work item agent:51 [1924250736] with exception [class io.cattle.platform.async.utils.TimeoutException:Timeout] 
2016-12-22 23:27:53,240 INFO  [:] [] [] [] [ecutorService-1] [i.c.p.c.v.impl.ConfigUpdatePublisher]         Finished work item agent:51 [1077578467] with exception [class io.cattle.platform.async.utils.TimeoutException:Timeout] 
2016-12-22 23:27:53,240 INFO  [:] [] [] [] [ecutorService-1] [i.c.p.c.v.impl.ConfigUpdatePublisher]         Finished work item agent:51 [23370807] with exception [class io.cattle.platform.async.utils.TimeoutException:Timeout] 
2016-12-22 23:27:53,246 INFO  [:] [] [] [] [ecutorService-1] [i.c.p.c.v.impl.ConfigUpdatePublisher]         Finished work item agent:51 [241250137] with exception [class io.cattle.platform.async.utils.TimeoutException:Timeout] 
```


After about 40 mts , all most all of them(17 out of 20) have come to "Running" state.

```
validrc02-1dc32               0/1       ContainerCreating   0          37m
validrc02-2i0jm               1/1       Running             0          37m
validrc02-5mo68               1/1       Running             0          37m
validrc02-8reev               1/1       Running             0          37m
validrc02-cbtx4               1/1       Running             0          37m
validrc02-cig46               1/1       Running             0          37m
validrc02-clhep               1/1       Running             0          37m
validrc02-g4wxw               1/1       Running             0          37m
validrc02-h4n4q               1/1       Running             0          37m
validrc02-jddqw               1/1       Running             0          37m
validrc02-jgry1               0/1       ContainerCreating   0          37m
validrc02-kh231               0/1       ContainerCreating   0          37m
validrc02-kt8ge               1/1       Running             0          37m
validrc02-lm2t8               1/1       Running             0          37m
validrc02-mdcij               1/1       Running             0          37m
validrc02-oppcd               1/1       Running             0          37m
validrc02-oy630               1/1       Running             0          37m
validrc02-uivxd               1/1       Running             0          37m
validrc02-vntyw               1/1       Running             0          37m
validrc02-xfer8               1/1       Running             0          37m
```

**Comment by sangeethah** [comment]
There was this 1 issue seen when testing with  v1.2.2-rc3 - Delete  large number (200) offending pods that were in bad state (constant restarts). Creating pods after this was stuck in "ContainerCreating" state for a very long time.

This issue was fixed in v1.2.2-rc4.
